### PR TITLE
Add CPU test for Nano GPT Model

### PR DESCRIPTION
**Pull request by pdeviTT**

### CODE REVIEW

**Comment by pdeviTT** [comment]
@vkovinicTT @mstojkovicTT  , Can you please review this patch :)

File: forge/test/models/pytorch/text/nanogpt/test_nanogpt.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+@pytest.mark.nightly
+@pytest.mark.parametrize("variant", ["FinancialSupport/NanoGPT"])
+def test_nanogpt_test_generation(record_forge_property, variant):
+
+    # Build Module Name
+    module_name = build_module_name(
+        framework=Framework.PYTORCH,
+        model="nanogpt",
+        variant=variant,
+        task=Task.SPEECH_RECOGNITION,
```
Comment (mstojkovicTT) [inline forge/test/models/pytorch/text/nanogpt/test_nanogpt.py:34]: Is there a reason for using `SPEECH_RECOGNITION` instead of `TEXT_GENERATION`? If it is intentional, a short comment clarifying the usage might help avoid confusion later :))

File: forge/test/models/pytorch/text/nanogpt/test_nanogpt.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+    # Input prompt
+    input_prompt = "The financial market showed signs of volatility"
+
+    # Tokenize input
+    inputs = tokenizer(
+        input_prompt,
+        return_tensors="pt",
+        max_length=150,
+        pad_to_max_length=True,
```
Comment (mstojkovicTT) [inline forge/test/models/pytorch/text/nanogpt/test_nanogpt.py:54]: In newer versions of Hugging Face Transformers, `pad_to_max_length=True` is deprecated. The alternative to that is `padding=True` :))

File: forge/test/models/pytorch/text/nanogpt/test_nanogpt.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        truncation=True,
+    )
+    input_ids = inputs["input_ids"]
+    attn_mask = inputs["attention_mask"]
+
+    # Get Inputs
+    input_ids = input_ids.to(torch.int32)
+    attn_mask = attn_mask.to(torch.float32)
```
Comment (mstojkovicTT) [inline forge/test/models/pytorch/text/nanogpt/test_nanogpt.py:62]: Converting the attention mask to `torch.float32` might be nonstandard since attention masks are often integer (0 or 1) values. If this conversion is required, consider adding a comment to explain the reasoning so future readers understand the choice :))

File: forge/test/models/pytorch/text/nanogpt/test_nanogpt.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        return_tensors="pt",
+        max_length=150,
+        pad_to_max_length=True,
+        truncation=True,
+    )
+    input_ids = inputs["input_ids"]
+    attn_mask = inputs["attention_mask"]
+
+    # Get Inputs
+    input_ids = input_ids.to(torch.int32)
```
Comment (mstojkovicTT) [inline forge/test/models/pytorch/text/nanogpt/test_nanogpt.py:61]: Converting input_ids to int32 is a bit wierd, is there a reason for that?

**Review by mstojkovicTT — CHANGES_REQUESTED** [review]

File: forge/test/models/pytorch/text/nanogpt/test_nanogpt.py (hunk 1 above)
Comment (pdeviTT) [reply forge/test/models/pytorch/text/nanogpt/test_nanogpt.py:34]: I will update it for text generation

**Comment by pdeviTT** [comment]
Updated this patch. Can you please review it again @mstojkovicTT :))

**Review by vkovinicTT — APPROVED** [review]
Just change status of this PR from draft to open.

**Comment by pdeviTT** [comment]
Hi @vkovinicTT @mstojkovicTT , Not able to merge this patch. Can you please confirm on your side :))
